bc_obps/reporting/api/report_person_responsible.py

```diff
@@ -0,0 +1,44 @@
+from typing import Literal, Tuple
+from common.permissions import authorize
+from django.http import HttpRequest
+from registration.decorators import handle_http_errors
+from reporting.constants import EMISSIONS_REPORT_TAGS
+from reporting.schema.generic import Message
+from service.report_person_responsible import ReportContactService
+from service.error_service.custom_codes_4xx import custom_codes_4xx
+from .router import router
+from ..models import ReportPersonResponsible
+from ..schema.report_person_responsible import ReportPersonResponsibleIn, ReportPersonResponsibleOut
+
+
+@router.get(
+    "/report-version/{version_id}/person-responsible",
+    response={200: ReportPersonResponsibleOut, custom_codes_4xx: Message},
+    tags=EMISSIONS_REPORT_TAGS,
+    description="""Takes version_id (primary key of Report_Version model) and returns its report_operation object.""",
+    auth=authorize("approved_authorized_roles"),
+)
+@handle_http_errors()
+def get_report_person_responsible_by_version_id(
+    request: HttpRequest, version_id: int
+) -> Tuple[Literal[200], ReportPersonResponsibleOut]:
+    report_person_responsible = ReportContactService.get_report_person_responsible_by_version_id(version_id)
+    return 200, report_person_responsible  # type: ignore
+
+
+@router.post(
+    "/report-version/{version_id}/report-contact",
+    response={201: ReportPersonResponsibleOut, custom_codes_4xx: Message},
+    tags=EMISSIONS_REPORT_TAGS,
+    description=(
+        """Creates or updates a contact associated with a report version.
+                    Includes fields like legal name, trade name, operation details, and contact information."""
+    ),
+    auth=authorize("approved_authorized_roles"),
```

Comment:
This permissions should be scoped to the `approved_industry_user` as an idir user should not be able to mutate the contents of a report.
```suggestion
    auth=authorize("approved_industry_user"),
```